File: src/main/java/net/atos/client/bag/util/StatusWoonplaatsEnumAdapter.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.bag.util;

import jakarta.json.bind.adapter.JsonbAdapter;

import net.atos.client.bag.model.StatusWoonplaats;

public class StatusWoonplaatsEnumAdapter implements JsonbAdapter<StatusWoonplaats, String> {

    @Override
    public String adaptToJson(final StatusWoonplaats statusWoonplaats) {
        return statusWoonplaats.toString();
    }

    @Override
    public StatusWoonplaats adaptFromJson(final String json) {
        return StatusWoonplaats.fromValue(json);
    }
}

File: src/main/kotlin/nl/info/client/bag/util/StatusWoonplaatsEnumAdapter.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.bag.util

import jakarta.json.bind.adapter.JsonbAdapter
import nl.info.client.bag.model.generated.StatusWoonplaats

class StatusWoonplaatsEnumAdapter : JsonbAdapter<StatusWoonplaats, String> {
    override fun adaptToJson(statusWoonplaats: StatusWoonplaats): String = statusWoonplaats.toString()
    override fun adaptFromJson(json: String): StatusWoonplaats = StatusWoonplaats.fromValue(json)
}
